File: tusow/src/main/java/it/unibo/coordination/tusow/exceptions/UnauthorizedError.java
package it.unibo.coordination.tusow.exceptions;

public class UnauthorizedError extends ClientError {

    private static final int CODE = 401;
    private static final String MSG = "Unauthorized";

    public UnauthorizedError() {
        super(CODE, MSG);
    }

    public UnauthorizedError(String message) {
        super(CODE, MSG + ": " + message);
    }

    public UnauthorizedError(String message, Throwable cause) {
        super(CODE, MSG + ": " +  message, cause);
    }

    public UnauthorizedError(Throwable cause) {
        super(CODE, MSG, cause);
    }
}

File: tusow-service/src/main/kotlin/it/unibo/coordination/tusow/exceptions/UnauthorizedError.kt
package it.unibo.coordination.tusow.exceptions

class UnauthorizedError : ClientError {
    constructor() : super(CODE, MSG)
    constructor(message: String) : super(CODE, "$MSG: $message")
    constructor(message: String, cause: Throwable?) : super(CODE, "$MSG: $message", cause)
    constructor(cause: Throwable?) : super(CODE, MSG, cause)

    companion object {
        private const val CODE = 401
        private const val MSG = "Unauthorized"
    }
}